### ISSUE TITLE
spectrum.CIESession.set_response MemoryError: Unable to allocate array with shape (59999, 59999) and data type float64

### ISSUE THREAD

**Issue by ogoann** [AtomDB/pyatomdb#20]
- platform: Scientific Linux 7.7, python 3.7.3
- version: _newsession_ branch

I am seeing the following error: 

``` python
import pyatomdb
import numpy

cie = pyatomdb.spectrum.CIESession()
ebins = numpy.linspace(0.1,10,60000)
cie.set_response(ebins, raw=True)

---------------------------------------------------------------------------
MemoryError                               Traceback (most recent call last)
<ipython-input-3-560ee3b91f5b> in <module>
      1 cie = adb.spectrum.CIESession()
      2 ebins = np.linspace(0.1,10,60000)
----> 3 cie.set_response(ebins, raw=True)

/local/data/lemssd/amogorza/local/src/anaconda3/lib/python3.7/site-packages/pyatomdb-0.7.2-py3.7-linux-x86_64.egg/pyatomdb/spectrum.py in set_response(self, rmf, arf, raw)
   1759       self.specbin_units='keV'
   1760 
-> 1761       self.rmfmatrix = numpy.zeros([len(rmf)-1, len(rmf)-1])
   1762       for i in range(len(rmf)-1):
   1763         self.rmfmatrix[i,i] = 1.0

MemoryError: Unable to allocate array with shape (59999, 59999) and data type float64
``` 

This seems to depend on the size of `ebins` , and I suppose is because the rmf gets too big? If that's the case, I'd perhaps check for the size of the response and throw an appropriate error message.  

The reason I have tried to use a large number of energy bins is that I wanted to look at a  theoretical spectrum with a large precision. Perhaps in this case creating such a large rmf matrix is not really necessary. I am unsure if many people would be interested in such functionality though.

**Comment by jagophile** [comment]
Yes this is a known issue and it is just running out of memory. 

Things I can think of:
  (1) making a special diagonal matrix response which doesn't actually make the matrix (and so saves memory)
  (2) accepting this a limit, but warning as you suggest.

I will look in to it

**Comment by jagophile** [comment]
This has now been worked on, and push to the newsession branch.

When a dummy spectrum is defined, instead of trying to create a diagonal NxN matrix, the response matrix is just not applied (so should not run out of memory in large cases).

I haven't changed the error message, but I'll add that under general list of enhancements requests - meaningful error messages...

Let me know if it works, and I'll merge into the main branch,

**Comment by jagophile** [comment]
OK, I haven't heard any complaints and it seems to work, so I am closing this. It's been merged into the main branch now, and included in release v0.8.0
